src/lib.rs

```diff
@@ -178,7 +178,7 @@ pub trait StreamingIterator {
         Self: Sized,
         F: FnMut(&Self::Item) -> bool,
     {
-        Filter { it: self, f: f }
+        Filter { it: self, f }
```

Comment:
I would rather the clippy changes be considered in a separate PR -- can you rebase without that commit?